lib/msal-common/src/authority/RegionDiscovery.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule } from "../network/INetworkModule";
+import { IMDSBadResponse } from "../response/IMDSBadResponse";
+
+export class RegionDiscovery {
+    // Network interface to make requests with.
+    protected networkInterface: INetworkModule;
+    // The IMDS endpoint to retrieve region information.
+    protected static IMDS_ENDPOINT = "http://169.254.169.254/metadata/instance/compute/location";
+    // Options for the IMDS endpoint request
+    protected static IMDS_OPTIONS = {headers: {"Metadata": "true"}};
+    // The version of the IMDS endpoint to use
+    protected static VERSION = "2020-06-01";
+    // Region variable name
+    protected static REGION_ENV_NAME = "REGION_NAME";
+
+    constructor(networkInterface: INetworkModule) {
+        this.networkInterface = networkInterface;
+    }
+
+    /**
+     * Detect the region from the application's environment.
+     * 
+     * @returns Promise<string | null>
+     */
+    public async detectRegion(): Promise<string | null> {
+        // Detect region from the process environment variable
+        let autodetectedRegionName = process.env[RegionDiscovery.REGION_ENV_NAME];
```

Comment:
This will not be available in browser, please find another way to get the region env or move to node

Reply:
@tnorling how do you propose we move the code to `msal-node` whilst being so tightly knit with the `Authority class`

Reply:
Can you include this value on `PreferredAzureRegionOptions`?

Reply:
@sameerag what's you thought on adding this value to the `PreferredAzureRegionOptions`, the implication would be that we are moving the region discovery from the environment variable to the consumer of the API.

Reply:
Pro:
- It allows them to get the environment variable in whatever way that suites their environment

Con:
- It shifts the responsibility to the users of the library

Reply:
You can still set it in `msal-node`, without user providing it if you like, since we are passing this value into authority instantiation. But we should not be putting code in common that is not supported in downstream libraries.

Reply:
Gotcha, that makes sense, thanks @tnorling, we'll work on that improvement

Reply:
@tnorling I have implemented the changes you suggested please re-review.